interfaces/zookeeper_client/v0/README.md

```diff
@@ -0,0 +1,56 @@
+# `zookeeper_client`
+
+## Usage
+
+This relation interface describes the expected behavior of any charm claiming to be able to interface with a Zookeeper cluster as a client.
+
+## Direction
+
+```mermaid
+flowchart TD
+    Requirer -- database, \nrequested-secrets --> Provider
+    Provider -- database, \nendpoints, \nsecret-user --> Requirer
+```
+
+## Behavior
+
+Both the Requirer and the Provider need to adhere to criteria to be considered compatible with the interface.
+
+### Provider
+
+- Is expected to create an application user inside the database cluster when the requirer provides the `database` field.
+- Is expected to provide credentials (`username` and `password`) in a Juju Secret whenever the Requirer supplies the `database` field.
+- Is expected to expose the Juju Secrets URI to the credentials through the `secret-user` field of the data bag.
+- Is expected to provide the `endpoints` field with a comma-separated list of server uris and zNode.
+- Is expected to provide the `database` field with the zNode that was actually created.
+- Is expected to provide the `version` field whenever database charm wants to communicate its database version.
+- Is expected to provide the CA chain in the `tls-ca` field of a Juju Secret, whenever the provider has TLS enabled (such as using the [TLS Certificates Operator](https://github.com/canonical/tls-certificates-operator)).
+- Is expected to share the TLS Juju Secret URI through the `secret-tls` field of the databag.
+- If the Requirer asks for additional secrets (via `requested-secrets`, see below) other than those stored in the `user` and `tls` secrets, Provider is expected to define a `secret-extra` field holding the URI of the Juju Secret containing all additional secret fields.
+
+### Requirer
+
+- Is expected to provide `requested-secrets`, which is a list of field names that are not to be exposed on the relation databag, but handled within Juju Secrets. It should be JSON parsable array of strings, and correspond to valid Juju Secret keys (i.e. alphanumerical characters with a potential '-' (dash) character). Secret fields must contain `username` and `password` (and `tls-ca` in case TLS is enabled).
+- Is expected to provide a zNode in the `database` field.
```

Comment:
Is there any value in providing a `database` field at all knowing it can get ignored?

Reply:
This is similar to what is done in other interfaces. The two reasons that come to mind as to why the database received might be different from the one asked for are (1) the name is already in use and the client cannot access it, and (2) the name is malformed (for instance omitting the leading / in the chroot zNode).